pkg/brokerpak/manifest.go

```diff
@@ -0,0 +1,155 @@
+// Copyright 2018 the Service Broker Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package brokerpak
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/GoogleCloudPlatform/gcp-service-broker/pkg/validation"
+	"github.com/GoogleCloudPlatform/gcp-service-broker/utils/stream"
+	"github.com/GoogleCloudPlatform/gcp-service-broker/utils/ziputil"
+
+	getter "github.com/hashicorp/go-getter"
+)
+
+const manifestName = "manifest.yml"
+
+type Manifest struct {
+	PackVersion int `yaml:"packversion" validate:"required,eq=1"`
+
+	Name               string              `yaml:"name" validate:"required"`
+	Version            string              `yaml:"version" validate:"required"`
+	Metadata           map[string]string   `yaml:"metadata"`
+	Platforms          []Platform          `yaml:"platforms" validate:"required,dive"`
+	TerraformResources []TerraformResource `yaml:"terraform_binaries" validate:"required,dive"`
+	ServiceDefinitions []string            `yaml:"service_definitions" validate:"required"`
+}
+
+func (m *Manifest) Validate() error {
+	return validation.ValidateStruct(m)
+}
+
+func (m *Manifest) AppliesToCurrentPlatform() bool {
+	for _, platform := range m.Platforms {
+		if platform.MatchesCurrent() {
+			return true
+		}
+	}
+
+	return false
+}
+
+func (m *Manifest) Pack(base, dest string) error {
+	dir, err := ioutil.TempDir("", "brokerpak")
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(dir) // clean up
+
+	if err := m.packSources(dir); err != nil {
+		return err
+	}
+
+	if err := m.packBinaries(dir); err != nil {
+		return err
+	}
+
+	if err := m.packDefinitions(dir, base); err != nil {
+		return err
+	}
+
+	return ziputil.Archive(dir, dest)
+}
+
+func (m *Manifest) packSources(tmp string) error {
+	for _, resource := range m.TerraformResources {
+		destination := filepath.Join(tmp, "src", resource.Name+".zip")
+		if err := fetchArchive(resource.Source, destination); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (m *Manifest) packBinaries(tmp string) error {
+	for _, platform := range m.Platforms {
+		platformPath := filepath.Join(tmp, "bin", platform.Os, platform.Arch)
+		for _, resource := range m.TerraformResources {
+			if err := getter.GetAny(platformPath, resource.Url(platform)); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func (m *Manifest) packDefinitions(tmp, base string) error {
+	manifestCopy := *m
+
+	var servicePaths []string
```

Comment:
Can you leave a code comment explaining the reasoning for moving the definition files from their original location to the `definitions` directory?